app/upgrades/v4/upgrade_test.go

```diff
@@ -0,0 +1,105 @@
+	"github.com/stretchr/testify/suite"
+	abci "github.com/tendermint/tendermint/abci/types"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+
+	"github.com/dymensionxyz/dymension/v3/app"
+	"github.com/dymensionxyz/dymension/v3/app/apptesting"
+)
+
+// UpgradeTestSuite defines the structure for the upgrade test suite
+type UpgradeTestSuite struct {
+	suite.Suite
+	Ctx sdk.Context
+	App *app.App
+}
+
+// SetupTest initializes the necessary items for each test
+func (s *UpgradeTestSuite) SetupTest(t *testing.T) {
+	s.App = apptesting.Setup(t, false)
+	s.Ctx = s.App.BaseApp.NewContext(false, tmproto.Header{Height: 1, ChainID: "dymension_100-1", Time: time.Now().UTC()})
+}
+
+// TestUpgradeTestSuite runs the suite of tests for the upgrade handler
+func TestUpgradeTestSuite(t *testing.T) {
+	suite.Run(t, new(UpgradeTestSuite))
+}
+
+var (
+	dummyUpgradeHeight                    int64 = 5
+	expectDelayedackDeletePacketBatchSize int32 = 1000
+	expectDelayedackEpochIdentifier             = "hour"
+	expectDelayedackBridgingFee                 = sdk.NewDecWithPrec(1, 3)
+)
+
+// TestUpgrade is a method of UpgradeTestSuite to test the upgrade process.
+func (s *UpgradeTestSuite) TestUpgrade() {
+	testCases := []struct {
+		msg         string
+		upgrade     func()
+		postUpgrade func() error
+		expPass     bool
+	}{
+		{
+			"Test that upgrade does not panic and sets correct parameters",
+
+			func() {
+				// Run upgrade
+				s.Ctx = s.Ctx.WithBlockHeight(dummyUpgradeHeight - 1)
+				plan := upgradetypes.Plan{Name: "v4", Height: dummyUpgradeHeight}
+				err := s.App.UpgradeKeeper.ScheduleUpgrade(s.Ctx, plan)
+				s.Require().NoError(err)
+				_, exists := s.App.UpgradeKeeper.GetUpgradePlan(s.Ctx)
+				s.Require().True(exists)
+
+				s.Ctx = s.Ctx.WithBlockHeight(dummyUpgradeHeight)
+				// simulate the upgrade process not panic.
+				s.Require().NotPanics(func() {
+					defer func() {
+						if r := recover(); r != nil {
+							s.Fail("Upgrade panicked", r)
+						}
+					}()
+					// simulate the upgrade process.
+					s.App.BeginBlocker(s.Ctx, abci.RequestBeginBlock{})
+				})
+			},
+			func() error {
+				// Post-update validation to ensure parameters are correctly set
+
+				// Check Delayedack parameters
+				delayedackParams := s.App.DelayedAckKeeper.GetParams(s.Ctx)
+				if delayedackParams.DeletePacketBatchSize != expectDelayedackDeletePacketBatchSize ||
+					delayedackParams.EpochIdentifier != expectDelayedackEpochIdentifier ||
+					!delayedackParams.BridgingFee.Equal(expectDelayedackBridgingFee) {
+					return fmt.Errorf("delayedack parameters not set correctly")
+				}
+
+				return nil
+			},
+			true,
+		},
+	}
+
+	for _, tc := range testCases {
+		s.Run(fmt.Sprintf("Case %s", tc.msg), func() {
+			s.SetupTest(s.T()) // Reset for each case
+
+			tc.upgrade()
+			err := tc.postUpgrade()
+			if tc.expPass {
+				s.Require().NoError(err)
+			} else {
+				s.Require().Error(err)
+			}
+		})
+	}
+}
```

Comment:
IMO this test is not well suited to a table test since there is only 1 test right now, and there are only 10 lines in the for loop and 40 lines in the list item

Just write a regular test